### PR TITLE
Added a makefile (and a .pc file for pkg-config)

### PR DESCRIPTION
**Pull request by mpevnev**
For those who prefer the traditional "make; make install" routine.

### CODE REVIEW

**Comment by BBasile** [comment]
At priori i have nothing against adding this but i have to say that using a makefile to setup source + binaries in system directories is quite uncommon.

File: dyaml.pc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+prefix=/usr
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: libdyaml
+Description: YAML parser and emitter library for D language
+Version: 0.4.5
```
Comment (BBasile) [inline dyaml.pc]: The version is not correct. It's 0.6.1 now

**Comment by mpevnev** [comment]
It was mostly meant as help for those willing to package the library into a specific Linux distro package format, and to ensure that dyaml.pc is relevant to the actual installation.

**Comment by mpevnev** [comment]
Actually, now that I've thought about it a bit more, for a static library this just doesn't really make sense. If it were an executable, or a dynamic library, then it would have some merit, but since it's not there's no point in it. I'm closing the pull request, sorry for bothering you.

**Comment by wilzbach** [comment]
> Actually, now that I've thought about it a bit more, for a static library this just doesn't really make sense. If it were an executable, or a dynamic library, then it would have some merit, but since it's not there's no point in it. I'm closing the pull request, sorry for bothering you.

No worries. I am not sure about your use case, but it sounds very close to a `dub install` option:

https://github.com/dlang/dub/issues/839

**Comment by ximion** [comment]
Just FTR, the biggest problem with dub in distro packaging is they way it handles (local) dependencies, and not really the invocation of dub or installing stuff to the right place (although support for all of that makes packaging a lot easier).

**Comment by ZombineDev** [comment]
@ximion can you explain this in more detail (or point me somewhere where I can read more on this)? Because as someone who uses dub, but has almost no understanding of the challanges related distro packaging, I don't see any issues with the way dub manages things. I think spreading such knowledge is very important, especially for third-party library developers as their stuff sooner or later gets used by an application like Tilix that people want to see packaged in their distro.
I would rather fix the problem in dub, than workaround it by adding a new build system for every project. Of course, in the short-term I understand why adding Meson as an alternative is important for you and I support your efforts.

**Comment by ximion** [comment]
@ZombineDev Sorry for the delayed reply... I made https://gist.github.com/ximion/fe6264481319dd94c8308b1ea4e8207a a while back to summarize some obstacles for packaging D stuff in distributions, not everything on this list is essential though.

The one thing that completely kills dub usage for distribution packages in all distributions though is that there is no way to create a local repository of dub packages that software consumes without being modified.
A Linux distribution has to be self-contained, all parts needed to build its components must be shipped as packages, fetching things from the internet is not allowed (we need to ensure the builds work in 10 years still, and ensure reproducibility of the builds).
So, essentially bug https://github.com/dlang/dub/issues/838 makes using dub impossible in distros (I tried, with some hacks, but it really doesn't work).

Meson on the other hand was built as a replacement for Autotools in GNOME and other core components of the Linux ecosystem (systemd and X are using it now), so it naturally takes care of integrating well with the existing ecosystem, since it had to provide equivalent features to Automake in order to be successful (which a language-specific build tool like dub doesn't necessarily need to).

**Pull request closed without merging**